### PR TITLE
acceptance: fix sporadic DNS problems

### LINKED ISSUE #4894: circleci: docker flakiness
Every now and then (but often enough to fail a significant amount of builds due to the sheer volume of iterations this bug gets), Gossip doesn't join up:
```
I160306 00:51:43.699152 cli/start.go:129  [build] alpha.v1-676-g3325b63 @ 2016/03/06 00:37:29 (go1.6)
I160306 00:51:43.701589 server/context.go:223  1 storage engine initialized
I160306 00:51:43.701841 cli/start.go:143  starting cockroach node
E160306 00:51:43.896722 gossip/gossip.go:733  invalid bootstrap address: &{typ:tcp addr:roach0.local:26257 exhausted:false}, lookup roach0.local: no such host
I160306 00:51:43.900384 storage/engine/rocksdb.go:106  opening rocksdb instance at "/data2.0"
I160306 00:51:43.909866 server/node.go:353  store store=0:0 ([]=/data2.0) not bootstrapped
I160306 00:51:43.910160 storage/stores.go:289  read 0 node addresses from persistent storage
I160306 00:51:43.911156 server/node.go:486  connecting to gossip network to verify cluster ID...
```

Note `E160306 00:51:43.896722 gossip/gossip.go:733  invalid bootstrap address: &{typ:tcp addr:roach0.local:26257 exhausted:false}, lookup roach0.local: no such host` which suggests that there's maybe a DNS issue with `docker-spy` and a subsequent failure in Gossip to retry in a speedy manner.

Repros locally, for example

```
while [ $? -eq 0 ]; do rm -rf ./acceptance/Test*; make acceptance TESTS=GossipPeerings TESTFLAGS='-v -l .'; done
```

### PR DESCRIPTION
**Pull request by petermattis**
Specify GODEBUG=netdns=cgo to work around sporadic DNS problems within
acceptance tests. Cgo DNS seems to be more robust than Go DNS.

Fixes #4894.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4942)

### CODE REVIEW

**Comment by petermattis** [comment]
I'm not particularly happy doing this. I'd prefer to understand why the Go DNS resolver is failing.

File: acceptance/cluster/localcluster.go [hunk 1]
```diff
@@ -363,6 +363,7 @@ func (l *LocalCluster) createRoach(node *testNode, vols *Container, cmd ...strin
 				defaultTCP: {},
 			},
 			Entrypoint: entrypoint,
+			Env:        []string{"GODEBUG=netdns=cgo"},
```
Comment (tamird) [inline acceptance/cluster/localcluster.go]: Add a TODO here if you intend to (or wish someone else would) dig into this.

**Comment by tamird** [comment]
LGTM

**Comment by tschottdorf** [comment]
LGTM

File: acceptance/cluster/localcluster.go (hunk 1 above)
Comment (petermattis) [inline acceptance/cluster/localcluster.go]: Done.

**Pull request merged**